libs/core/algorithms/include/hpx/parallel/algorithms/detail/find.hpp

```diff
@@ -34,8 +34,7 @@ namespace hpx::parallel::detail {
             sequential_find_t<ExPolicy>, Iterator first, Sentinel last,
             T const& value, Proj proj = Proj())
         {
-            return util::loop_pred<
-                std::decay_t<hpx::execution::sequenced_policy>>(
+            return util::loop_pred<ExPolicy>(
```

Comment:
Do you need to decay the `ExPolicy` here (and in other places)?

Reply:
Do we really need decay over here? ExPolicy is only being used for deducing if seq or unseq overload should be called. is_unseq<unseq_policy> should be true for both value and reference right?

Also, please note that this is branched from [simd-find](https://github.com/STEllAR-GROUP/hpx/pull/6302).

Reply:
Well, you tell me. I was just asking the question without closer investigation.

Reply:
To the best of my knowledge, I don't believe we need to decay. I am unsure why it was being decayed previously.

Reply:
@hkaiser I have tried checking behaviour with/without std::decay_t. The correct overloads are called in both cases. I am unsure why std::decay_t is being used.